### PR TITLE
Add llamaindex agent integration

### PR DESCRIPTION
**Pull request by colombod**
Adding contrib support for integrating llamaidnex agentrunner in autogen architecture

### CODE REVIEW

File: .devcontainer/Dockerfile
Comment (ekzhu) [inline .devcontainer/Dockerfile:1]: Not sure why this file is changed. Can we reset this one?

File: .devcontainer/devcontainer.json
Comment (ekzhu) [inline .devcontainer/devcontainer.json:1]: Let reset changes to this file.

File: .devcontainer/studio/Dockerfile
Comment (ekzhu) [inline .devcontainer/studio/Dockerfile:1]: let's reset this file.

File: .devcontainer/studio/devcontainer.json
Comment (ekzhu) [inline .devcontainer/studio/devcontainer.json:1]: let's reset this file

File: test/agentchat/contrib/test_llamaindex_conversable_agent.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    from llama_index.llms.openai import OpenAI
+
+    skip_for_dependencies = False
+except ImportError as e:
+    skip_for_dependencies = True
+    skip_reasons.append(f"dependency not installed: {e.msg}")
+    pass
+
+
+openaiKey = os.environ.get("OPENAI_API_KEY", "")
```
Comment (ekzhu) [inline test/agentchat/contrib/test_llamaindex_conversable_agent.py:29]: I think we use Azure Open AI in CI. You can take a look at what other tests does that read configuration from a file. I think you can also ready the key from that file.

File: test/test_notebook.py
Comment (ekzhu) [inline test/test_notebook.py:1]: Let's reset this file.

File: website/blog/authors.yml
Comment (ekzhu) [inline website/blog/authors.yml:1]: Let's reset this file.

**Comment by ekzhu** [comment]
cc @jerryjliu @Disiok

File: test/agentchat/contrib/test_llamaindex_conversable_agent.py (hunk 1 above)
Comment (ekzhu) [reply test/agentchat/contrib/test_llamaindex_conversable_agent.py:29]: I would consider remove OpenAI endpoint access dependency for this test. Just use pytest patch to check if the corresponding llamaindex functions are triggered. We don't need to unit test llama index functionality.

**Review by ekzhu — CHANGES_REQUESTED** [review]

File: .devcontainer/Dockerfile
Comment (colombod) [reply .devcontainer/Dockerfile:1]: trying very hard, is CLRF -> LF things

**Review by ekzhu — APPROVED** [review]
Nice. Thanks for the PR!

**Comment by lmcmahi** [comment]
Hi,
I am not able to import from autogen.agentchat.contrib.llamaindex_conversable_agent the LLamaIndexConversableAgent,
is there any separate package to install ? I am using 0.2.28

**Pull request merged**